src/libstore/path-info.hh

```diff
@@ -100,8 +100,8 @@ struct ValidPathInfo
 
     ValidPathInfo(const ValidPathInfo & other) = default;
 
-    ValidPathInfo(StorePath && path) : path(std::move(path)) { };
-    ValidPathInfo(const StorePath & path) : path(path) { };
+    ValidPathInfo(StorePath && path) : path(std::move(path)), narHash(Hash::dummy) { };
+    ValidPathInfo(const StorePath & path) : path(path), narHash(Hash::dummy) { };
```

Comment:
Ideally we'd require a `narHash` at construction time. Do you know if there are any places where we construct a `ValidPathInfo` and we don't know the `narHash`?

Reply:
Agreed. We should make this an argument, and at the very least callers would opt into using `Hash::dummy` and then initialize later, though better yet `Hash::dummy` can go away altogether.